source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,237 @@
+
+For example, include the following code to connect to a standalone MongoDB deployment running on localhost on port
+``27017`` and define ``database`` to refer to the ``test`` database and ``collection`` to refer to the ``restaurants``
+collection.
+
+.. code-block:: java
+
+   public class AggTourOne {
+
+     public static void main(String[] args) {
+       MongoClient mongoClient = MongoClients.create();
+       MongoDatabase database = mongoClient.getDatabase("test");
+       MongoCollection<Document> collection = database.getCollection("restaurants");
+
+
+       }
+   }
+
+.. tip::
+
+   For information on connecting to MongoDB, see the :ref:`Connection Guide <mongoclient>`
+
+Perform Aggregation
+~~~~~~~~~~~~~~~~~~~
+
+To perform an aggregation, pass a list of :manual:`aggregation stages </meta/aggregation-quick-reference/#stages>` to
+the `MongoCollection.aggregate() <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/MongoCollection.html#aggregate(com.mongodb.client.ClientSession,java.util.List,java.lang.Class)>`__
+method.
+
+The Java driver provides the `Aggregates <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html>`__
+helper class that
+contains builders for aggregation stages.
+
+In the following example, the aggregation pipeline:
+
+- Uses a :manual:`$match <reference/operator/aggregation/match/>` stage to filter for documents whose
+  ``categories`` array field contains the element ``Bakery``. The example uses
+  `Aggregates.match <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#match(org.bson.conversions.Bson)>`__
+  to build the ``$match`` stage.
+
+- Uses  a :manual:`$group <reference/operator/aggregation/group/>` stage to group the matching documents by the ``stars``
+  field, accumulating a count of documents for each distinct value of ``stars``.
+
+  The example uses `Aggregates.group <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#group(org.bson.conversions.Bson)>`__
+  to build the ``$group`` stage and `Accumulators.sum <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators#sum(java.lang.String,TExpression)>`__
+  to build the accumulator expression. The :manual:`accumulator expressions <reference/operator/aggregation/#accumulators-group>`
+  used within the ``$group`` stage are built with the `Accumulators <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators.html>`__
+  helper class.
+
+.. code-block:: java
+
+   collection.aggregate(
+         Arrays.asList(
+                 Aggregates.match(Filters.eq("categories", "Bakery")),
+                 Aggregates.group("$stars", Accumulators.sum("count", 1))
+         )
+   ).forEach(printBlock);
+
```

Comment:
I like to separate the output from sample code by explaining that running the code should produce the following output. Random example (authored by Katie) https://github.com/mongodb/docs-java/blob/master/source/fundamentals/crud/read-operations/limit.txt#L67
Would you agree with this or otherwise, is it not a good idea?

Reply:
I felt it was self-evident, but perhaps that's an incorrect position